### PR TITLE
deprecate generateGasPrice in favor of snakecased generate_gas_price

### PR DESCRIPTION
**Pull request by tmckenzie51**
### What was wrong?
Added generate_gas_price, deprecated generateGasPrice

Related to Issue #1429

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: docs/web3.eth.rst [hunk 1]
```diff
@@ -970,13 +970,18 @@ The following methods are available on the ``web3.eth`` namespace.
...
         20000000000
 
     .. note::
         For information about how gas price can be customized in web3 see
         :ref:`Gas_Price`.
 
+.. py:method:: Eth.generateGasPrice(transaction_params=None)
+
+    .. warning:: Deprecated: This method is deprecated in favor of
+      :meth:`~web3.eth.Eth.generate_gas_price_strategy()`
```
Comment (kclowes) [inline docs/web3.eth.rst:983]: copy/pasta :)
```suggestion
      :meth:`~web3.eth.Eth.generate_gas_price()`
```
```suggestion
      :meth:`~web3.eth.Eth.generate_gas_price_strategy()`
```

File: docs/gas_price.rst [hunk 2]
```diff
@@ -16,11 +16,11 @@ Retrieving gas price
 --------------------
 
 To retreive the gas price using the selected strategy simply call
-:meth:`~web3.eth.Eth.generateGasPrice`
+:meth:`~web3.eth.Eth.generate_gas_price`
 
 .. code-block:: python
 
-    >>> Web3.eth.generateGasPrice()
+    >>> Web3.eth.generate_gas_price()
```
Comment (kclowes) [inline docs/gas_price.rst:23]: This should be lowercase `w` in Web3:
```suggestion
    >>> web3.eth.generate_gas_price()
```
```suggestion
    >>> Web3.eth.generate_gas_price()
```

File: docs/web3.eth.rst [hunk 3]
```diff
@@ -970,13 +970,18 @@ The following methods are available on the ``web3.eth`` namespace.
 
     .. code-block:: python
 
-        >>> Web3.eth.generateGasPrice()
+        >>> Web3.eth.generate_gas_price()
```
Comment (kclowes) [inline docs/web3.eth.rst:973]: Same here re. lowercased 'w': 
```suggestion
        >>> web3.eth.generate_gas_price()
```
```suggestion
        >>> Web3.eth.generate_gas_price()
```

File: tests/core/eth-module/test_gas_pricing.py [hunk 4]
```diff
@@ -14,14 +14,14 @@ def test_get_set_gasPrice(web3):
 
 
 def test_no_gas_price_strategy_returns_none(web3):
```
Comment (kclowes) [inline tests/core/eth-module/test_gas_pricing.py:16]: Is this the one we paired on trying to add a deprecation warning? If not, I'd expect a test that ensures the `DeprecationWarning` gets raised somewhere in this file.

**Review by kclowes — APPROVED** [review]
Manual testing looks good to me! I just left a few doc comments, and one on testing for a `DeprecationWarning`.

File: docs/web3.eth.rst (hunk 1 above)
Comment (tmckenzie51) [reply docs/web3.eth.rst]: Oops, lol. Thanks!

**Pull request merged**